File: problems/517_super_washing_machines/solution.py
from typing import List


class Solution:
    def findMinMoves(self, machines: List[int]) -> int:
        total = sum(machines)
        n = len(machines)
        if total % n != 0:
            return -1
        avg = total // n
        res = 0
        running_sum = 0
        for m in machines:
            diff = m - avg
            running_sum += diff
            res = max(res, abs(running_sum), diff)
        return res

File: problems/517_super_washing_machines/Solution.java
class Solution {
    public int findMinMoves(int[] machines) {
        int total = 0;
        for (int m : machines) total += m;
        int n = machines.length;
        if (total % n != 0) return -1;
        int avg = total / n, res = 0, runningSum = 0;
        for (int m : machines) {
            int diff = m - avg;
            runningSum += diff;
            res = Math.max(res, Math.max(Math.abs(runningSum), diff));
        }
        return res;
    }
}
